docs/source/references/schema.json

```
@@ -1000,7 +1166,17 @@
       }
     ]
   },
-  "ExecutorEventSet": "u32",
+  "ExecutorEventSet": {
+    "Bitmap": {
+      "repr": "u32",
+      "masks": [
+        {
+          "name": "Upgraded",
+          "mask": 1
+        }
+      ]
+    }
+  },
```

```suggestion
  "ExecutorEventSet": {
    "Bitmap": {
      "masks": [
        {
          "name": "Upgraded",
          "mask": 1
        }
      ]
    }
  },
```

I don't think it's necessary to define `repr`. This implementation detail and doesn't belong to a schema

---

It is necessary to know to encode/decode SCALE.

---

then how come we don't have `repr` for enums?

---

I see, we implicitly assume `u8`

---

In the SCALE specification it is said that enum always take a single byte for discriminant.

> A fixed number of variants, each mutually exclusive and potentially implying a further value or series of values. **Encoded as the first byte identifying the index of the variant that the value is.** Any further bytes are used to encode any data that the variant implies. **Thus, no more than 256 variants are supported.**

https://docs.substrate.io/reference/scale-codec/

---

I guess we can fix it as a `u32`, I just wanted to give possible leeway in case the `BitmapMeta` is going to be reused for something that is not an `EventSet`

---

I think it's fine